src/CodeGen_Hexagon.cpp

```diff
@@ -508,6 +508,58 @@ llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(llvm::Function *intrin, Ty
         arg_types.insert(arg_types.begin() + 1, split_type);
     }
 
+    if (flags & HvxIntrinsic::InterleaveByteAndBroadcastToWord) {
+        internal_assert(args.size() >= 2);
+        vector<Value *> bytes(args.end()-2, args.end());
+        llvm::Function *fn = module->getFunction("halide.hexagon.interleave.b.dup2.h");
+        args[args.size()-2] =  builder->CreateCall(fn, { bytes[0], bytes[1] });
+        args.pop_back();
+    } else if (flags & HvxIntrinsic::ConcatHalves) {
+        internal_assert(args.size() >= 2);
+        vector<Value *> halves(args.end()-2, args.end());
+        llvm::Function *fn = module->getFunction("halide.hexagon.concat.h");
+        args[args.size()-2] =  builder->CreateCall(fn, { halves[0], halves[1] });
+        args.pop_back();
+    }
+    if (flags & HvxIntrinsic::ConcatVectorOps01) {
+        internal_assert(args.size() >= 2);
+        internal_assert(args[0]->getType()->isVectorTy());
+        internal_assert(args[1]->getType()->isVectorTy());
+
+        bool is_128B = target.has_feature(Halide::Target::HVX_128);
```

Comment:
I think this should be just

    args[0] = concat_vectors(args[0], args[1]);

If they aren't native vectors, this code is not correct. It's also a lot less code.